internal/guest/storage/devicemapper/devicemapper.go

```diff
@@ -269,14 +276,27 @@ func CreateDevice(name string, flags CreateFlags, targets []Target) (_ string, e
 }
 
 // RemoveDevice removes a device-mapper device and its associated device node.
-func RemoveDevice(name string) error {
-	f, err := openMapper()
-	if err != nil {
-		return err
+func RemoveDevice(name string) (err error) {
+	rm := func() error {
+		f, err := openMapperWrapper()
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		os.Remove(path.Join("/dev/mapper", name))
+		return removeDeviceWrapper(f, name)
 	}
-	defer f.Close()
-	os.Remove(path.Join("/dev/mapper", name))
-	return removeDevice(f, name)
+
+	// This is workaround for "device or resource busy" error, which occasionally happens after the device mapper
+	// target has been unmounted.
+	for i := 0; i < 10; i++ {
+		if err = rm(); err != nil && err == syscall.EBUSY {
```

Comment:
Should've caught this when I made the last comment but the err != nil check is redundant, if it's EBUSY it's not nil so just that portion of the check should be enough. If it's some other error or nil then we want to exit this loop anyways.

Reply:
duh, I'm tunnel visioning this jeez